src/compiler/transformers/esDecorators.ts

```diff
@@ -2332,4 +2340,59 @@ export function transformESDecorators(context: TransformationContext): (x: Sourc
             factory.createStringLiteral("default") :
             factory.createStringLiteralFromNode(name);
     }
+
+    function createMetadata(name: Identifier, hasProto: boolean) {
+        const varDecl = factory.createVariableDeclaration(
+            name,
+            /*exclamationToken*/ undefined,
+            /*type*/ undefined,
+            factory.createConditionalExpression(
+                factory.createBinaryExpression(

```

Comment:
We have a shorthand for this:
```suggestion
                factory.createLogicalAnd(
```